app/src/test/resources/design-document.md

```diff
@@ -0,0 +1,151 @@
+# Design Document
+
+## _Collector's Stash_ Design
+
+## 1. Problem Statement
+Many individuals have a passion for collecting various and unique items, be it comic books, stamps, or other collectibles. They often resort to carrying physical notepads and books to manually document and keep track of their collections. This traditional method is overwhelming and can also be prone to errors, loss, and inefficiency. To address this issue, I am developing a innovative application aimed at providing collectors with a digital solution for managing and preserving their collections.
+
+## Top Questions to Resolve in Review
+1. How many tables do I need?
+2. Do I need to use a GSI (Let's say I use it so they can filter/search for their favorites or such)
+3. What is the main goal vs the stretch goals, but clear on what the end goal looks like realistically
+
+## Use Cases
+U1. _As a user, I would like to store my collectibles_
+u2. _As a user, I would like to be able to view the books in my collection_
+u3. _As a user, I would like to favorite the books I like most_
+U4. _As a user, I would like to view to store the price I paid for such book_
+U5. _As a user, I would like to select a series I am collection and see the missing books_
+U6. _As a user, I would like to update my collection with new books or collectibles I have purchased_
+U7. _As a user, I would like to update my collection, to get rid of books I might have sold off and no longer own_
+U8. _As a user, I would like add a book to a new series I have started collection_
+U9. _As a user, I would like create an account_
+U10. _As a user, I would like to login to my account_
+U12. _As a user, I would like to see collection based of my login_
+
+Stretch goals:
+- Search Option
+- Adding Images to the comics added(if wanted, user takes a picture and able to store that image)
+- Better design on website
+
+## Project Scope
+- view books that have been added to my collection
+- add books to a new series I am collection
+- view all books in my collection
+
+## Proposed Architecture Overview
+We will use API Gateway and Lambda to create endpoints()
+(GetAllComicBooksLambda, GetAllFavoriteLambda, GetAllStampsLambda, GetAllCoinsLambda, AddComicBookToCollectionLambda, AddStampsToCollectionLambda, AddCoinsToCollectionLambda, UpdateComicBookCollectionLambda, UpdateStampCollectionLambda, UpdateCoinCollectionLambda, SearchCollectionLambda)
+_that will handle the creation, update, and retrieval of events and vendors to satisfy our requirements_
+
+## API
+
+## 6.1 Public Models
+```
+// ComicBookModel
+
+String id
+String volumeNumber
+String issueNumber
+String date
+Int price
+Boolean isFavorite
+String publisher
+```
+// Can these Coins and Stamp use same data table?
+// If not use different tables
+```
+// CoinAndStampModel
+
+String id
+String year
+String Condition
+Int price
+String Country
+Boolean isFavorite
+```
```

Comment:
Are you thinking of supporting multiple types of collectibles, or focusing on books for the MVP? 

You know I'm a fan of starting with a minimal MVP. If you are wanting to focus on books, it looks like this section and the API endpoints will need to be updated. (And db table design added.)

Reply:
Changed from Coins, Stamps, And Comic books to just comic  books, so updated endpoints and tables